--- FFMpeg-20050806/libavcodec/avcodec.h.orig	2005-08-01 22:07:03.000000000 +0200
+++ FFMpeg-20050806/libavcodec/avcodec.h	2005-12-31 10:40:05.280500500 +0100
@@ -16,7 +16,7 @@
 
 //FIXME the following 2 really dont belong in here
 #define FFMPEG_VERSION_INT     0x000409
-#define FFMPEG_VERSION         "CVS"
+#define FFMPEG_VERSION         "49CVS"
 
 #define AV_STRINGIFY(s)	AV_TOSTRING(s)
 #define AV_TOSTRING(s) #s
--- FFMpeg-20050806/configure.orig	2005-12-31 09:31:31.000000000 +0100
+++ FFMpeg-20050806/configure	2005-12-31 10:40:54.511577250 +0100
@@ -1691,8 +1691,8 @@
 
 # build pkg-config files libavcodec.pc, libavformat.pc and libpostproc.pc 
 
-lavc_build=`grep '#define LIBAVCODEC_BUILD' "$source_path/libavcodec/avcodec.h" | sed 's/[^0-9]//g'`
-lavf_build=`grep '#define LIBAVFORMAT_BUILD' "$source_path/libavformat/avformat.h" | sed 's/[^0-9]//g'`
+lavc_build=`egrep '#define LIBAVCODEC_VERSION +[0-9\.]' "$source_path/libavcodec/avcodec.h" | sed 's/[^0-9]//g'`
+lavf_build=`egrep '#define LIBAVFORMAT_VERSION +[0-9\.]' "$source_path/libavformat/avformat.h" | sed 's/[^0-9]//g'`
 
 requires=
 test "$libogg" = "yes" && requires="$requires ogg >= 1.1"
