--- ffmpeg-0.4.6/libavcodec/Makefile.orig	Sun Jan  5 02:32:23 2003
+++ ffmpeg-0.4.6/libavcodec/Makefile	Sun Jan  5 02:49:11 2003
@@ -10,33 +10,33 @@
 CFLAGS= $(OPT) -Wall -DHAVE_AV_CONFIG_H -I.. -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_GNU_SOURCE
 LDFLAGS= $(LDOPT)
 
-OBJS= common.o utils.o mem.o allcodecs.o \
-      mpegvideo.o h263.o jrevdct.o jfdctfst.o jfdctint.o\
-      mpegaudio.o ac3enc.o mjpeg.o resample.o dsputil.o \
-      motion_est.o imgconvert.o imgresample.o msmpeg4.o \
-      mpeg12.o h263dec.o svq1.o rv10.o mpegaudiodec.o pcm.o simple_idct.o \
-      ratecontrol.o adpcm.o eval.o dv.o error_resilience.o \
-      wmadec.o fft.o mdct.o mace.o huffyuv.o
+OBJS= common.lo utils.lo mem.lo allcodecs.lo \
+      mpegvideo.lo h263.lo jrevdct.lo jfdctfst.lo jfdctint.lo\
+      mpegaudio.lo ac3enc.lo mjpeg.lo resample.lo dsputil.lo \
+      motion_est.lo imgconvert.lo imgresample.lo msmpeg4.lo \
+      mpeg12.lo h263dec.lo svq1.lo rv10.lo mpegaudiodec.lo pcm.lo simple_idct.lo \
+      ratecontrol.lo adpcm.lo eval.lo dv.lo error_resilience.lo \
+      wmadec.lo fft.lo mdct.lo mace.lo huffyuv.lo
 ASM_OBJS=
 
 # currently using liba52 for ac3 decoding
 ifeq ($(CONFIG_AC3),yes)
-OBJS+= a52dec.o
+OBJS+= a52dec.lo
 
 # using builtin liba52 or runtime linked liba52.so.0
 ifneq ($(CONFIG_A52BIN),yes)
-OBJS+= liba52/bit_allocate.o liba52/bitstream.o liba52/downmix.o \
-	liba52/imdct.o  liba52/parse.o
+OBJS+= liba52/bit_allocate.lo liba52/bitstream.lo liba52/downmix.lo \
+	liba52/imdct.lo  liba52/parse.lo
 endif
 endif
 
 ifeq ($(CONFIG_MP3LAME),yes)
-OBJS += mp3lameaudio.o
+OBJS += mp3lameaudio.lo
 EXTRALIBS += -lmp3lame
 endif
 
 ifeq ($(CONFIG_VORBIS),yes)
-OBJS += oggvorbis.o
+OBJS += oggvorbis.lo
 EXTRALIBS += -lvorbis -lvorbisenc
 endif
 
@@ -47,52 +47,54 @@
 
 # i386 mmx specific stuff
 ifeq ($(TARGET_MMX),yes)
-OBJS += i386/fdct_mmx.o i386/cputest.o \
-	i386/dsputil_mmx.o i386/mpegvideo_mmx.o \
-	i386/idct_mmx.o i386/motion_est_mmx.o \
-	i386/simple_idct_mmx.o
+OBJS += i386/fdct_mmx.lo i386/cputest.lo \
+	i386/dsputil_mmx.lo i386/mpegvideo_mmx.lo \
+	i386/idct_mmx.lo i386/motion_est_mmx.lo \
+	i386/simple_idct_mmx.lo
 endif
 
 # armv4l specific stuff
 ifeq ($(TARGET_ARCH_ARMV4L),yes)
-ASM_OBJS += armv4l/jrevdct_arm.o
-OBJS += armv4l/dsputil_arm.o armv4l/mpegvideo_arm.o
+ASM_OBJS += armv4l/jrevdct_arm.lo
+OBJS += armv4l/dsputil_arm.lo armv4l/mpegvideo_arm.lo
 endif
 
 # sun mediaLib specific stuff
 # currently only works when libavcodec is used in mplayer
 ifeq ($(HAVE_MLIB),yes)
-OBJS += mlib/dsputil_mlib.o
+OBJS += mlib/dsputil_mlib.lo
 CFLAGS += $(MLIB_INC)
 endif
 
 # alpha specific stuff
 ifeq ($(TARGET_ARCH_ALPHA),yes)
-OBJS += alpha/dsputil_alpha.o alpha/mpegvideo_alpha.o \
-	alpha/simple_idct_alpha.o alpha/motion_est_alpha.o
-ASM_OBJS += alpha/dsputil_alpha_asm.o alpha/motion_est_mvi_asm.o
+OBJS += alpha/dsputil_alpha.lo alpha/mpegvideo_alpha.lo \
+	alpha/simple_idct_alpha.lo alpha/motion_est_alpha.lo
+ASM_OBJS += alpha/dsputil_alpha_asm.lo alpha/motion_est_mvi_asm.lo
 CFLAGS += -fforce-addr -freduce-all-givs
 endif
 
 ifeq ($(TARGET_ARCH_POWERPC),yes)
-OBJS += ppc/dsputil_ppc.o ppc/mpegvideo_ppc.o
+OBJS += ppc/dsputil_ppc.lo ppc/mpegvideo_ppc.lo
 endif
 
 ifeq ($(TARGET_MMI),yes)
-OBJS += ps2/dsputil_mmi.o ps2/idct_mmi.o ps2/mpegvideo_mmi.o
+OBJS += ps2/dsputil_mmi.lo ps2/idct_mmi.lo ps2/mpegvideo_mmi.lo
 endif
 
 ifeq ($(TARGET_ALTIVEC),yes)
 CFLAGS += -faltivec
-OBJS += ppc/dsputil_altivec.o ppc/mpegvideo_altivec.o ppc/idct_altivec.o
+OBJS += ppc/dsputil_altivec.lo ppc/mpegvideo_altivec.lo ppc/idct_altivec.lo
 endif
 
-SRCS := $(OBJS:.o=.c) $(ASM_OBJS:.o=.S)
+SRCS := $(OBJS:.lo=.c) $(ASM_OBJS:.lo=.S)
 OBJS := $(OBJS) $(ASM_OBJS)
 
-LIB= libavcodec.a
+LIB= libavcodec.la
 ifeq ($(BUILD_SHARED),yes)
-SLIB= libavcodec.so
+RPATH= -rpath $(prefix)/lib libavcodec.so
+else
+RPATH=
 endif
 TESTS= imgresample-test dct-test motion-test fft-test
 
@@ -101,38 +103,33 @@
 tests: apiexample cpuid_test $(TESTS)
 
 $(LIB): $(OBJS)
-	rm -f $@
-	$(AR) rc $@ $(OBJS)
-	$(RANLIB) $@
+	libtool --mode=link --tag CXX $(CC) $(LDFLAGS) -o $@ $(OBJS) $(RPATH) $(EXTRALIBS) -release $(VERSION)
 
-$(SLIB): $(OBJS)
-	$(CC) $(SHFLAGS) -o $@ $(OBJS) $(EXTRALIBS)
-
-dsputil.o: dsputil.c dsputil.h
+dsputil.lo: dsputil.c dsputil.h
 
 # specific sse code
-%_sse.o : %_sse.c
-	$(CC) $(CFLAGS) -msse -c -o $@ $< 
+%_sse.lo : %_sse.c
+	libtool --mode=compile  --tag CXX $(CC) $(CFLAGS) -msse -c -o $@ $< 
 
 # specific 3dnow code
-%_3dnow.o : %_3dnow.c
-	$(CC) $(CFLAGS) -m3dnow -c -o $@ $< 
+%_3dnow.lo : %_3dnow.c
+	libtool --mode=compile --tag CXX  $(CC) $(CFLAGS) -m3dnow -c -o $@ $< 
 
-%.o: %.c
-	$(CC) $(CFLAGS) -c -o $@ $< 
+%.lo: %.c
+	libtool --mode=compile --tag CXX $(CC) $(CFLAGS) -c -o $@ $< 
 
-%.o: %.S
-	$(CC) $(CFLAGS) -c -o $@ $<
+%.lo: %.S
+	libtool --mode=compile --tag CXX $(CC) $(CFLAGS) -c -o $@ $<
 
 # motion_est_alpha uses the MVI extension, which is not available with
 # -mcpu=ev4 (default) or ev5/ev56. Thus, force -mcpu=pca56 in those
 # cases.
 ifeq ($(TARGET_ARCH_ALPHA),yes)
-alpha/motion_est_alpha.o: alpha/motion_est_alpha.c
+alpha/motion_est_alpha.lo: alpha/motion_est_alpha.c
 	cpu=`echo "$(CFLAGS)" | sed -n 's,.*-mcpu=\([a-zA-Z0-9]*\).*,\1,p'`; \
 	case x"$$cpu" in x|xev[45]*) newcpu=pca56;; *) newcpu=$$cpu;; esac; \
 	echo $(CC) $(CFLAGS) -mcpu=$$newcpu -c -o $@ $<;\
-	$(CC) $(CFLAGS) -mcpu=$$newcpu -c -o $@ $<
+	libtool --mode=compile --tag CXX $(CC) $(CFLAGS) -mcpu=$$newcpu -c -o $@ $<
 endif
 
 # depend only used by mplayer now
@@ -142,13 +139,13 @@
 	$(CC) -MM $(CFLAGS) $(SRCS) 1>.depend
 
 clean: 
-	rm -f *.o *~ .depend $(LIB) $(SLIB) *.so i386/*.o i386/*~ \
-	   armv4l/*.o armv4l/*~ \
-	   mlib/*.o mlib/*~ \
-	   alpha/*.o alpha/*~ \
-	   ppc/*.o ppc/*~ \
-	   ps2/*.o ps2/*~ \
-	   liba52/*.o liba52/*~ \
+	rm -f *.*o *~ .depend $(LIB) $(SLIB) *.so *.la i386/*.*o i386/*~ \
+	   armv4l/*.*o armv4l/*~ \
+	   mlib/*.*o mlib/*~ \
+	   alpha/*.*o alpha/*~ \
+	   ppc/*.*o ppc/*~ \
+	   ps2/*.*o ps2/*~ \
+	   liba52/*.*o liba52/*~ \
 	   apiexample $(TESTS)
 
 distclean: clean
@@ -156,43 +153,41 @@
 
 # api example program
 apiexample: apiexample.c $(LIB)
-	$(CC) $(CFLAGS) -o $@ $< $(LIB) $(EXTRALIBS) -lm
+	libtool --mode=link  --tag CXX $(CC) $(LDFLAGS) $(CFLAGS) -o $@ $< $(LIB) $(EXTRALIBS) -lm
 
 # cpuid test
 cpuid_test: i386/cputest.c
-	$(CC) $(CFLAGS) -D__TEST__ -o $@ $<
+	libtool --mode=link --tag CXX  $(CC) $(LDFLAGS) $(CFLAGS) -D__TEST__ -o $@ $<
 
 # testing progs
 
 imgresample-test: imgresample.c
-	$(CC) $(CFLAGS) -DTEST -o $@ $^ -lm
+	libtool --mode=link --tag CXX  $(CC) $(LDFLAGS) $(CFLAGS) -DTEST -o $@ $^ -lm
 
 dct-test: dct-test.o jfdctfst.o jfdctint.o i386/fdct_mmx.o\
           fdctref.o jrevdct.o i386/idct_mmx.o simple_idct.o i386/simple_idct_mmx.o
-	$(CC) -o $@ $^ -lm
+	libtool --mode=link --tag CXX  $(CC) $(LDFLAGS) -o $@ $^ -lm
 
 motion-test: motion_test.o $(LIB)
-	$(CC) -o $@ $^ -lm
+	libtool --mode=link --tag CXX  $(CC) $(LDFLAGS) -o $@ $^ -lm
 
 fft-test: fft-test.o fft.o mdct.o
-	$(CC) -o $@ $^ -lm
+	libtool --mode=link --tag CXX  $(CC) $(LDFLAGS) -o $@ $^ -lm
 
 install: all
 ifeq ($(BUILD_SHARED),yes)
-	install -d $(prefix)/lib
-	install -s -m 755 $(SLIB) $(prefix)/lib/libavcodec-$(VERSION).so
-	ln -sf libavcodec-$(VERSION).so $(prefix)/lib/libavcodec.so
-	ldconfig || true
-	mkdir -p $(prefix)/include/ffmpeg
-	install -m 644 $(VPATH)/avcodec.h $(prefix)/include/ffmpeg/avcodec.h
-	install -m 644 $(VPATH)/common.h $(prefix)/include/ffmpeg/common.h
+	install -d $(DESTDIR)$(prefix)/lib
+	libtool --mode=install install $(LIB) $(DESTDIR)$(prefix)/lib
+	install -d $(DESTDIR)$(prefix)/include/ffmpeg
+	install -m 644 $(VPATH)/avcodec.h $(DESTDIR)$(prefix)/include/ffmpeg/avcodec.h
+	install -m 644 $(VPATH)/common.h $(DESTDIR)$(prefix)/include/ffmpeg/common.h
 endif
 
 installlib: all
-	install -m 644 $(LIB) $(prefix)/lib
-	mkdir -p $(prefix)/include/ffmpeg
+	libtool --mode=install install $(LIB) $(DESTDIR)$(prefix)/lib
+	install -d $(DESTDIR)$(prefix)/include/ffmpeg
 	install -m 644 $(SRC_PATH)/libavcodec/avcodec.h $(SRC_PATH)/libavcodec/common.h \
-                $(prefix)/include/ffmpeg
+                $(DESTDIR)$(prefix)/include/ffmpeg
 
 #
 # include dependency files if they exist
--- ffmpeg-0.4.6/libavformat/Makefile.orig	Sun Jan  5 02:32:23 2003
+++ ffmpeg-0.4.6/libavformat/Makefile	Sun Jan  5 02:53:50 2003
@@ -8,65 +8,66 @@
 
 CFLAGS= $(OPT) -Wall -I.. -I$(SRC_PATH) -I$(SRC_PATH)/libavcodec -DHAVE_AV_CONFIG_H -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_GNU_SOURCE
 
-OBJS= utils.o cutils.o allformats.o
+OBJS= utils.lo cutils.lo allformats.lo
 
 # mux and demuxes
-OBJS+=mpeg.o mpegts.o ffm.o crc.o img.o raw.o rm.o asf.o \
-      avienc.o avidec.o wav.o swf.o au.o gif.o mov.o jpeg.o dv.o framehook.o
+OBJS+=mpeg.lo mpegts.lo ffm.lo crc.lo img.lo raw.lo rm.lo asf.lo \
+      avienc.lo avidec.lo wav.lo swf.lo au.lo gif.lo mov.lo jpeg.lo dv.lo framehook.lo
 # file I/O
-OBJS+= avio.o aviobuf.o file.o 
+OBJS+= avio.lo aviobuf.lo file.lo 
 
 ifeq ($(BUILD_STRPTIME),yes)
-OBJS+= strptime.o
+OBJS+= strptime.lo
 endif
 
 ifeq ($(CONFIG_VIDEO4LINUX),yes)
-OBJS+= grab.o
+OBJS+= grab.lo
 endif
 
 ifeq ($(CONFIG_AUDIO_OSS),yes)
-OBJS+= audio.o 
+OBJS+= audio.lo 
 endif
 
 ifeq ($(CONFIG_AUDIO_BEOS),yes)
-OBJS+= beosaudio.o
+OBJS+= beosaudio.lo
 endif
 
 ifeq ($(CONFIG_NETWORK),yes)
-OBJS+= udp.o tcp.o http.o rtsp.o rtp.o rtpproto.o
+OBJS+= udp.lo tcp.lo http.lo rtsp.lo rtp.lo rtpproto.lo
 # BeOS network stuff
 ifeq ($(NEED_INET_ATON),yes)
-OBJS+= barpainet.o
+OBJS+= barpainet.lo
 endif
 endif
 
 ifeq ($(CONFIG_VORBIS),yes)
-OBJS+= ogg.o
+OBJS+= ogg.lo
 endif
 
-LIB= libavformat.a
+LIB= libavformat.la
 
 all: $(LIB)
 
 $(LIB): $(OBJS)
-	rm -f $@
-	$(AR) rc $@ $(OBJS)
-	$(RANLIB) $@
-
-installlib: all
-	install -m 644 $(LIB) $(prefix)/lib
-	mkdir -p $(prefix)/include/ffmpeg
-	install -m 644 $(SRC_PATH)/libav/avformat.h $(SRC_PATH)/libav/avio.h \
-                $(SRC_PATH)/libav/rtp.h $(SRC_PATH)/libav/rtsp.h \
-                $(SRC_PATH)/libav/rtspcodes.h \
-                $(prefix)/include/ffmpeg
+	libtool --mode=link --tag CXX $(CC) $(LDOPT) -o $@ $(OBJS) ../libavcodec/libavcodec.la -rpath $(prefix)/lib -release $(VERSION)
+
+install: all
+	install -d $(DESTDIR)$(prefix)/lib
+	libtool --mode=install install $(LIB) $(DESTDIR)$(prefix)/lib
+	install -d $(DESTDIR)$(prefix)/include/ffmpeg
+	install -m 644 $(SRC_PATH)/libavformat/avformat.h $(SRC_PATH)/libavformat/avio.h \
+                $(SRC_PATH)/libavformat/rtp.h $(SRC_PATH)/libavformat/rtsp.h \
+                $(SRC_PATH)/libavformat/rtspcodes.h \
+                $(DESTDIR)$(prefix)/include/ffmpeg
+
+installlib: install
 
-%.o: %.c
-	$(CC) $(CFLAGS) -c -o $@ $< 
+%.lo: %.c
+	libtool --mode=compile --tag CXX $(CC) $(CFLAGS) -c -o $@ $< 
 
 # BeOS: remove -Wall to get rid of all the "multibyte constant" warnings
 %.o: %.cpp
-	g++ $(subst -Wall,,$(CFLAGS)) -c -o $@ $< 
+	libtool --mode=compile --tag CXX g++ $(subst -Wall,,$(CFLAGS)) -c -o $@ $< 
 
 clean: 
-	rm -f *.o *~ *.a 
+	rm -f *.o *.lo *~ *.a *.la *.so
--- ffmpeg-0.4.6/Makefile.orig	Sun Jan  5 02:32:23 2003
+++ ffmpeg-0.4.6/Makefile	Sun Jan  5 03:01:25 2003
@@ -19,7 +19,7 @@
 PROG=ffmpeg$(EXE)
 else
 EXT=
-PROG=ffmpeg ffplay
+PROG=ffmpeg
 ifeq ($(CONFIG_FFSERVER),yes)
 PROG+=ffserver
 endif
@@ -30,7 +30,7 @@
 endif
 
 ifeq ($(BUILD_SHARED),yes)
-DEP_LIBS=libavcodec/libavcodec.so libavformat/libavformat.a
+DEP_LIBS=libavcodec/libavcodec.la libavformat/libavformat.la
 else
 DEP_LIBS=libavcodec/libavcodec.a libavformat/libavformat.a
 ifeq ($(CONFIG_MP3LAME),yes)
@@ -56,17 +56,12 @@
 	$(MAKE) -C libavcodec all
 	$(MAKE) -C libavformat all
 
-ffmpeg_g$(EXE): ffmpeg.o $(DEP_LIBS)
-	$(CC) $(LDFLAGS) -o $@ ffmpeg.o -L./libavcodec -L./libavformat \
-              -lavformat -lavcodec $(EXTRALIBS)
-
-ffmpeg$(EXE): ffmpeg_g$(EXE)
-	cp -p $< $@ ; $(STRIP) $@
+ffmpeg$(EXE): ffmpeg.o $(DEP_LIBS)
+	libtool --mode=link  --tag CXX $(CC) $(LDFLAGS) -o $@ ffmpeg.o $(DEP_LIBS) $(EXTRALIBS)
 
 ffserver$(EXE): ffserver.o $(DEP_LIBS)
-	$(CC) $(LDFLAGS) $(FFSLDFLAGS) \
-		-o $@ ffserver.o -L./libavcodec -L./libavformat \
-              -lavformat -lavcodec $(EXTRALIBS) 
+	libtool --mode=link --tag CXX $(CC) $(LDFLAGS) $(FFSLDFLAGS) \
+		-o $@ ffserver.o $(DEP_LIBS) $(EXTRALIBS)
 
 ffplay: ffmpeg$(EXE)
 	ln -sf $< $@
@@ -79,14 +74,15 @@
 
-install: all $(INSTALLVHOOK)
+install: $(INSTALLVHOOK)
 	$(MAKE) -C libavcodec install
-	install -d $(prefix)/bin
-	install -s -m 755 $(PROG) $(prefix)/bin
-	ln -sf ffmpeg $(prefix)/bin/ffplay 
+	$(MAKE) -C libavformat install
+	install -d $(DESTDIR)$(prefix)/bin
+	libtool --mode=install install $(PROG) $(DESTDIR)$(prefix)/bin
+	ln -sf ffmpeg $(DESTDIR)$(prefix)/bin/ffplay 
 
-install-vhook: $(prefix)/lib/vhook
-	$(MAKE) -C vhook install INSTDIR=$(prefix)/lib/vhook
+install-vhook: $(DESTDIR)$(prefix)/lib/vhook
+	$(MAKE) -C vhook install INSTDIR=$(DESTDIR)$(prefix)/lib/vhook
 
-$(prefix)/lib/vhook:
+$(DESTDIR)$(prefix)/lib/vhook:
 	install -d $@
 
 installlib:
