--- ffmpeg/libavcodec/libx264.c.org	2008-09-30 08:13:47.298284337 +0200
+++ ffmpeg/libavcodec/libx264.c	2008-09-30 08:16:20.034103090 +0200
@@ -162,7 +162,11 @@
 
     x4->params.i_bframe = avctx->max_b_frames;
     x4->params.b_cabac = avctx->coder_type == FF_CODER_TYPE_AC;
+#if X264_BUILD > 60
     x4->params.i_bframe_adaptive = avctx->b_frame_strategy;
+#else
+    x4->params.b_bframe_adaptive = avctx->b_frame_strategy;
+#endif
     x4->params.i_bframe_bias = avctx->bframebias;
     x4->params.b_bframe_pyramid = avctx->flags2 & CODEC_FLAG2_BPYRAMID;
     avctx->has_b_frames= avctx->flags2 & CODEC_FLAG2_BPYRAMID ? 2 : !!avctx->max_b_frames;
