diff -urN ffmpeg/Makefile ffmpeg.new/Makefile
--- ffmpeg/Makefile	Wed Aug 15 15:12:07 2001
+++ ffmpeg.new/Makefile	Wed Sep  5 13:51:41 2001
@@ -3,8 +3,8 @@
 #
 include config.mak
 
-CFLAGS= $(OPTFLAGS) -Wall -g -I./libavcodec -I./libav 
-LDFLAGS= -g 
+CFLAGS= $(OPT) -Wall -I./libavcodec -I./libav 
+LDFLAGS= $(LDOPT)
 ifeq ($(TARGET_GPROF),yes)
 CFLAGS+=-p
 LDFLAGS+=-p
@@ -19,16 +19,16 @@
 	$(MAKE) -C libav all
 
 ffmpeg: ffmpeg.o libav/libav.a libavcodec/libavcodec.a
-	gcc $(LDFLAGS) -o $@ $^ -lm
+	$(CC) $(LDFLAGS) -o $@ $^ -lm
 
 ffserver: ffserver.o libav/libav.a libavcodec/libavcodec.a
-	gcc $(LDFLAGS) -o $@ $^ -lm
+	$(CC) $(LDFLAGS) -o $@ $^ -lm
 
 %.o: %.c
-	gcc $(CFLAGS) -c -o $@ $< 
+	$(CC) $(CFLAGS) -c -o $@ $< 
 
 install: all
-	install -s -m 755 $(PROG) $(prefix)/bin
+	install -m 755 $(PROG) $(prefix)/bin
 
 clean: 
 	$(MAKE) -C libavcodec clean
diff -urN ffmpeg/libav/Makefile ffmpeg.new/libav/Makefile
--- ffmpeg/libav/Makefile	Wed Aug 15 00:32:50 2001
+++ ffmpeg.new/libav/Makefile	Wed Sep  5 13:47:25 2001
@@ -1,5 +1,5 @@
 include ../config.mak
-CFLAGS= $(OPTFLAGS) -Wall -g -I../libavcodec -DHAVE_AV_CONFIG_H
+CFLAGS= $(OPT) -Wall -I../libavcodec -DHAVE_AV_CONFIG_H
 
 OBJS= rm.o mpeg.o asf.o avienc.o jpeg.o swf.o wav.o raw.o \
 	 avidec.o ffm.o \
diff -urN ffmpeg/libavcodec/Makefile ffmpeg.new/libavcodec/Makefile
--- ffmpeg/libavcodec/Makefile	Mon Aug 13 00:52:36 2001
+++ ffmpeg.new/libavcodec/Makefile	Wed Sep  5 13:47:34 2001
@@ -1,7 +1,7 @@
 include ../config.mak
 
-CFLAGS= $(OPTFLAGS) -Wall -g -DHAVE_AV_CONFIG_H
-LDFLAGS= -g
+CFLAGS= $(OPT) -Wall -DHAVE_AV_CONFIG_H
+LDFLAGS= $(LDOPT)
 
 OBJS= common.o utils.o mpegvideo.o h263.o jrevdct.o jfdctfst.o \
       mpegaudio.o ac3enc.o mjpeg.o resample.o dsputil.o \
