--- ffmpeg-export-2006-12-04/libavcodec/imgconvert.c.orig	2006-12-21 17:28:49.000000000 +0000
+++ ffmpeg-export-2006-12-04/libavcodec/imgconvert.c	2006-12-21 17:30:07.000000000 +0000
@@ -2299,7 +2299,6 @@
     return 0;
 }
 
-#ifndef CONFIG_SWSCALER
 /* XXX: always use linesize. Return -1 if not supported */
 int img_convert(AVPicture *dst, int dst_pix_fmt,
                 const AVPicture *src, int src_pix_fmt,
@@ -2541,7 +2540,6 @@
     avpicture_free(tmp);
     return ret;
 }
-#endif
 
 /* NOTE: we scan all the pixels to have an exact information */
 static int get_alpha_info_pal8(const AVPicture *src, int width, int height)
