--- ffmpeg/libavcodec/i386/dsputil_mmx.c.orig	2006-01-12 23:43:17.000000000 +0100
+++ ffmpeg/libavcodec/i386/dsputil_mmx.c	2006-01-29 14:45:40.325739750 +0100
@@ -617,7 +617,7 @@
     );
 }
 
-static inline void transpose4x4(uint8_t *dst, uint8_t *src, int dst_stride, int src_stride){
+static  void transpose4x4(uint8_t *dst, uint8_t *src, int dst_stride, int src_stride){
     asm volatile( //FIXME could save 1 instruction if done as 8x4 ...
         "movd  %4, %%mm0                \n\t"
         "movd  %5, %%mm1                \n\t"
