--- ffmpeg-2.0/configure.orig	2013-07-14 11:29:51.533724995 +0200
+++ ffmpeg-2.0/configure	2013-07-14 11:59:02.196984862 +0200
@@ -4154,7 +4154,7 @@
 enabled libnut            && require libnut libnut.h nut_demuxer_init -lnut
 enabled libopencore_amrnb && require libopencore_amrnb opencore-amrnb/interf_dec.h Decoder_Interface_init -lopencore-amrnb
 enabled libopencore_amrwb && require libopencore_amrwb opencore-amrwb/dec_if.h D_IF_init -lopencore-amrwb
-enabled libopencv         && require_pkg_config opencv opencv/cxcore.h cvCreateImageHeader
+enabled libopencv         && add_extralibs -lopencv_core -lopencv_features2d -lopencv_imgproc -lopencv_objdetect -lgcrypt
 enabled libopenjpeg       && { check_lib openjpeg-1.5/openjpeg.h opj_version -lopenjpeg ||
                                check_lib openjpeg.h opj_version -lopenjpeg ||
                                die "ERROR: libopenjpeg not found"; }
