diff -durN -x '*~' ffmpeg-0.4.8.orig/Makefile ffmpeg-0.4.8/Makefile
--- ffmpeg-0.4.8.orig/Makefile	2004-12-29 13:19:12.000000000 +0000
+++ ffmpeg-0.4.8/Makefile	2004-12-29 13:21:21.234042184 +0000
@@ -111,10 +111,10 @@
 	fi
 endif
 
-install-vhook: $(DESTDIR)$(prefix)/lib/vhook
-	$(MAKE) -C vhook install INSTDIR=$(DESTDIR)$(prefix)/lib/vhook
+install-vhook: $(DESTDIR)$(prefix)/lib64/vhook
+	$(MAKE) -C vhook install INSTDIR=$(DESTDIR)$(prefix)/lib64/vhook
 
-$(DESTDIR)$(prefix)/lib/vhook:
+$(DESTDIR)$(prefix)/lib64/vhook:
 	install -d $@
 
 installlib:
diff -durN -x '*~' ffmpeg-0.4.8.orig/configure ffmpeg-0.4.8/configure
--- ffmpeg-0.4.8.orig/configure	2004-12-29 13:19:12.000000000 +0000
+++ ffmpeg-0.4.8/configure	2004-12-29 13:20:47.130226752 +0000
@@ -682,7 +682,7 @@
 EOF
 
 imlib2=no
-if $cc -I/usr/X11R6/include -L/usr/X11R6/lib -o $TMPE $TMPC -lImlib2 2> /dev/null  ; then
+if $cc -I/usr/X11R6/include -L/usr/X11R6/lib64 -o $TMPE $TMPC -lImlib2 2> /dev/null  ; then
 imlib2=yes
 fi
 
diff -durN -x '*~' ffmpeg-0.4.8.orig/libavcodec/Makefile ffmpeg-0.4.8/libavcodec/Makefile
--- ffmpeg-0.4.8.orig/libavcodec/Makefile	2004-12-29 13:19:12.000000000 +0000
+++ ffmpeg-0.4.8/libavcodec/Makefile	2004-12-29 13:22:35.653728672 +0000
@@ -141,7 +141,7 @@
 
 LIB= libavcodec.la
 ifeq ($(BUILD_SHARED),yes)
-RPATH= -rpath $(prefix)/lib
+RPATH= -rpath $(prefix)/lib64
 else
 RPATH=
 endif
@@ -221,15 +221,15 @@
 ifeq ($(CONFIG_WIN32),yes)
 	install -s -m 755 $(SLIB) "$(prefix)"
 else
-	install -d $(DESTDIR)$(prefix)/lib
-	libtool --mode=install install $(LIB) $(DESTDIR)$(prefix)/lib
+	install -d $(DESTDIR)$(prefix)/lib64
+	libtool --mode=install install $(LIB) $(DESTDIR)$(prefix)/lib64
 endif
 else
 install:
 endif
 
 installlib: all install-headers
-	libtool --mode=install install $(LIB) $(DESTDIR)$(prefix)/lib
+	libtool --mode=install install $(LIB) $(DESTDIR)$(prefix)/lib64
 
 install-headers:
 	install -d "$(DESTDIR)$(prefix)/include/ffmpeg"
diff -durN -x '*~' ffmpeg-0.4.8.orig/libavformat/Makefile ffmpeg-0.4.8/libavformat/Makefile
--- ffmpeg-0.4.8.orig/libavformat/Makefile	2004-12-29 13:19:12.000000000 +0000
+++ ffmpeg-0.4.8/libavformat/Makefile	2004-12-29 13:23:58.159185936 +0000
@@ -62,7 +62,7 @@
 
 LIB= libavformat.la
 ifeq ($(BUILD_SHARED),yes)
-RPATH= -rpath $(prefix)/lib
+RPATH= -rpath $(prefix)/lib64
 else
 RPATH=
 endif
@@ -82,15 +82,15 @@
 ifeq ($(CONFIG_WIN32),yes)
 	install -s -m 755 $(SLIB) "$(prefix)"
 else
-	install -d $(DESTDIR)$(prefix)/lib
-	libtool --mode=install install $(LIB) $(DESTDIR)$(prefix)/lib
+	install -d $(DESTDIR)$(prefix)/lib64
+	libtool --mode=install install $(LIB) $(DESTDIR)$(prefix)/lib64
 endif
 else
 install:
 endif
 
 installlib: all install-headers
-	libtool --mode=install install $(LIB) $(DESTDIR)$(prefix)/lib
+	libtool --mode=install install $(LIB) $(DESTDIR)$(prefix)/lib64
 
 install-headers:
 	install -d "$(DESTDIR)$(prefix)/include/ffmpeg"
