--- FFMpeg-20050806/configure.orig	2005-08-04 21:50:09.000000000 +0200
+++ FFMpeg-20050806/configure	2005-10-24 08:26:37.374034008 +0200
@@ -1703,7 +1703,7 @@
 cat <<EOF >libavcodec.pc
 prefix=$prefix
 exec_prefix=\${prefix}
-libdir=\${exec_prefix}/lib
+libdir=$libdir
 includedir=\${prefix}/include
 
 Name: libavcodec
@@ -1711,8 +1711,8 @@
 Version: $version-$lavc_build
 Requires: $requires
 Conflicts:
-Libs: -L\${libdir} -lavcodec $extralibs
-Cflags: -I\${includedir} -I\${includedir}/ffmpeg
+Libs: -L$libdir -lavcodec $extralibs
+Cflags: -I\${includedir}/ffmpeg
 EOF
 
 cat <<EOF >libavcodec-uninstalled.pc
