diff -Nur ffmpeg-0.4.8.orig/libavcodec/libpostproc/Makefile ffmpeg-0.4.8/libavcodec/libpostproc/Makefile
--- ffmpeg-0.4.8.orig/libavcodec/libpostproc/Makefile	2003-09-28 17:26:40.000000000 +0200
+++ ffmpeg-0.4.8/libavcodec/libpostproc/Makefile	2003-10-13 22:12:58.960929336 +0200
@@ -1,26 +1,21 @@
 
 include ../../config.mak
 
-ifeq ($(SHARED_PP),yes)
-SPPLIB = libpostproc.so
-SPPVERSION = 0.0.1
-endif
-PPLIB = libpostproc.a
+PPLIB = libpostproc.la
 
-PPOBJS=postprocess.o
-SPPOBJS=postprocess_pic.o
+PPOBJS=postprocess.lo
 
 CFLAGS  = $(OPTFLAGS) $(MLIB_INC) -I. -I.. $(EXTRA_INC)
 # -I/usr/X11R6/include/
 
 .SUFFIXES: .c .o
 
 # .PHONY: all clean
 
-.c.o:
-	$(CC) -c $(CFLAGS) -I.. -I../.. -o $@ $<
+%.lo: %.c
+	libtool --mode=compile $(CC) -c $(CFLAGS) -I.. -I../.. -o $@ $<
 
-all:    $(SWSLIB) $(PPLIB) $(SPPLIB)
+all:    $(SWSLIB) $(PPLIB)
 
 clean:
 	rm -f *.o *.a *~ *.so
@@ -34,27 +29,20 @@
 	$(CC) -MM $(CFLAGS) postprocess.c 1>.depend
 
 ifeq ($(SHARED_PP),yes)
-postprocess_pic.o: postprocess.c
-	$(CC) -c $(CFLAGS) -fomit-frame-pointer -fPIC -DPIC -I.. -I../.. -o $@ $<
-
-$(SPPLIB): $(SPPOBJS)
-	$(CC) -shared -Wl,-soname,$(SPPLIB).0 \
-	-o $(SPPLIB) $(SPPOBJS)
+RPATH= -rpath $(prefix)/lib
+else
+RPATH=
 endif
 
 $(PPLIB): $(PPOBJS)
-	rm -f $@
-	$(AR) rc $@ $(PPOBJS)
-	$(RANLIB) $@
+	libtool --mode=link $(CC) -o $(PPLIB) $(PPOBJS) $(RPATH) -version-info 0:1:0
 
 install: all
 ifeq ($(SHARED_PP),yes)
-	install -d $(prefix)/lib
-	install -s -m 755 $(SPPLIB) $(prefix)/lib/$(SPPLIB).$(SPPVERSION)
-	ln -sf $(SPPLIB).$(SPPVERSION) $(prefix)/lib/$(SPPLIB)
-	ldconfig || true
-	mkdir -p $(prefix)/include/postproc
-	install -m 644 postprocess.h $(prefix)/include/postproc/postprocess.h
+	install -d $(DESTDIR)$(prefix)/lib
+	libtool --mode=install install -m 755 $(PPLIB) $(DESTDIR)$(prefix)/lib
+	install -d  $(DESTDIR)$(prefix)/include/postproc
+	install -m 644 postprocess.h $(DESTDIR)$(prefix)/include/postproc/postprocess.h
 endif
 
 
diff -Nur ffmpeg-0.4.8.orig/libavcodec/Makefile ffmpeg-0.4.8/libavcodec/Makefile
--- ffmpeg-0.4.8.orig/libavcodec/Makefile	2003-09-28 17:26:39.000000000 +0200
+++ ffmpeg-0.4.8/libavcodec/Makefile	2003-10-13 22:25:12.418426848 +0200
@@ -9,24 +9,24 @@
 # NOTE: -I.. is needed to include config.h
 CFLAGS= $(OPTFLAGS) -Wall -DHAVE_AV_CONFIG_H -I.. -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_GNU_SOURCE
 
-OBJS= common.o utils.o mem.o allcodecs.o \
-      mpegvideo.o jrevdct.o jfdctfst.o jfdctint.o\
-      mpegaudio.o ac3enc.o mjpeg.o resample.o dsputil.o \
-      motion_est.o imgconvert.o imgresample.o \
-      mpeg12.o mpegaudiodec.o pcm.o simple_idct.o \
-      ratecontrol.o adpcm.o eval.o dv.o error_resilience.o \
-      fft.o mdct.o mace.o huffyuv.o cyuv.o opts.o raw.o h264.o golomb.o \
-      vp3.o asv1.o 4xm.o cabac.o ffv1.o ra144.o ra288.o vcr1.o cljr.o \
-      roqvideo.o dpcm.o interplayvideo.o xan.o
+OBJS= common.lo utils.lo mem.lo allcodecs.lo \
+      mpegvideo.lo jrevdct.lo jfdctfst.lo jfdctint.lo \
+      mpegaudio.lo ac3enc.lo mjpeg.lo resample.lo dsputil.lo \
+      motion_est.lo imgconvert.lo imgresample.lo \
+      mpeg12.lo mpegaudiodec.lo pcm.lo simple_idct.lo \
+      ratecontrol.lo adpcm.lo eval.lo dv.lo error_resilience.lo \
+      fft.lo mdct.lo mace.lo huffyuv.lo cyuv.lo opts.lo raw.lo h264.lo golomb.lo \
+      vp3.lo asv1.lo 4xm.lo cabac.lo ffv1.lo ra144.lo ra288.lo vcr1.lo cljr.lo \
+      roqvideo.lo dpcm.lo interplayvideo.lo xan.lo
 
 ifeq ($(AMR_NB),yes)
 ifeq ($(AMR_NB_FIXED),yes)
-OBJS+= amr.o
-AMREXTRALIBS+= amr/*.o
+OBJS+= amr.lo
+AMREXTRALIBS+= amr/*.lo
 AMRLIBS=amrlibs
 CLEANAMR=cleanamr
 else
-OBJS+= amr.o amr_float/sp_dec.o amr_float/sp_enc.o amr_float/interf_dec.o amr_float/interf_enc.o
+OBJS+= amr.lo amr_float/sp_dec.lo amr_float/sp_enc.lo amr_float/interf_dec.lo amr_float/interf_enc.lo
 CLEANAMR=cleanamrfloat
 endif
 endif
@@ -35,26 +35,26 @@
 
 # codecs which are patented in some non free countries like the us
 ifeq ($(CONFIG_RISKY),yes)
-OBJS+= h263.o msmpeg4.o h263dec.o svq1.o rv10.o wmadec.o indeo3.o
+OBJS+= h263.lo msmpeg4.lo h263dec.lo svq1.lo rv10.lo wmadec.lo indeo3.lo
 endif
 
 ifeq ($(HAVE_XVMC_ACCEL),yes)
-OBJS+= xvmcvideo.o
+OBJS+= xvmcvideo.lo
 endif
 
 # currently using liba52 for ac3 decoding
 ifeq ($(CONFIG_AC3),yes)
-OBJS+= a52dec.o
+OBJS+= a52dec.lo
 
 # using builtin liba52 or runtime linked liba52.so.0
 ifneq ($(CONFIG_A52BIN),yes)
-OBJS+= liba52/bit_allocate.o liba52/bitstream.o liba52/downmix.o \
-	liba52/imdct.o  liba52/parse.o liba52/crc.o liba52/resample.o
+OBJS+= liba52/bit_allocate.lo liba52/bitstream.lo liba52/downmix.lo \
+	liba52/imdct.lo  liba52/parse.lo liba52/crc.lo liba52/resample.lo
 endif
 endif
 
 ifeq ($(CONFIG_FAAD),yes)
-OBJS+= faad.o
+OBJS+= faad.lo
 ifeq ($(CONFIG_FAADBIN),yes)
 # no libs needed
 else
@@ -64,20 +64,20 @@
 
 ifeq ($(CONFIG_PP),yes)
 ifeq ($(SHARED_PP),yes)
-EXTRALIBS += -lpostproc
+EXTRALIBS += libpostproc/libpostproc.la
 else
 # LIBS += libpostproc/libpostproc.a ... should be fixed
-OBJS += libpostproc/postprocess.o
+OBJS += libpostproc/postprocess.lo
 endif
 endif
 
 ifeq ($(CONFIG_MP3LAME),yes)
-OBJS += mp3lameaudio.o
+OBJS += mp3lameaudio.lo
 EXTRALIBS += -lmp3lame
 endif
 
 ifeq ($(CONFIG_VORBIS),yes)
-OBJS += oggvorbis.o
+OBJS += oggvorbis.lo
 EXTRALIBS += -lvorbis -lvorbisenc
 endif
 
@@ -88,60 +88,62 @@
 
 # i386 mmx specific stuff
 ifeq ($(TARGET_MMX),yes)
-OBJS += i386/fdct_mmx.o i386/cputest.o \
-	i386/dsputil_mmx.o i386/mpegvideo_mmx.o \
-	i386/idct_mmx.o i386/motion_est_mmx.o \
-	i386/simple_idct_mmx.o i386/fft_sse.o
+OBJS += i386/fdct_mmx.lo i386/cputest.lo \
+	i386/dsputil_mmx.lo i386/mpegvideo_mmx.lo \
+	i386/idct_mmx.lo i386/motion_est_mmx.lo \
+	i386/simple_idct_mmx.lo i386/fft_sse.lo
 ifdef TARGET_BUILTIN_VECTOR
-i386/fft_sse.o: CFLAGS+= -msse
+i386/fft_sse.lo: CFLAGS+= -msse
 endif
 endif
 
 # armv4l specific stuff
 ifeq ($(TARGET_ARCH_ARMV4L),yes)
-ASM_OBJS += armv4l/jrevdct_arm.o armv4l/simple_idct_arm.o
-OBJS += armv4l/dsputil_arm.o armv4l/mpegvideo_arm.o
+ASM_OBJS += armv4l/jrevdct_arm.lo armv4l/simple_idct_arm.lo
+OBJS += armv4l/dsputil_arm.lo armv4l/mpegvideo_arm.lo
 endif
 
 # sun mediaLib specific stuff
 # currently only works when libavcodec is used in mplayer
 ifeq ($(HAVE_MLIB),yes)
-OBJS += mlib/dsputil_mlib.o
+OBJS += mlib/dsputil_mlib.lo
 CFLAGS += $(MLIB_INC)
 endif
 
 # alpha specific stuff
 ifeq ($(TARGET_ARCH_ALPHA),yes)
-OBJS += alpha/dsputil_alpha.o alpha/mpegvideo_alpha.o \
-	alpha/simple_idct_alpha.o alpha/motion_est_alpha.o
-ASM_OBJS += alpha/dsputil_alpha_asm.o alpha/motion_est_mvi_asm.o
+OBJS += alpha/dsputil_alpha.lo alpha/mpegvideo_alpha.lo \
+	alpha/simple_idct_alpha.lo alpha/motion_est_alpha.lo
+ASM_OBJS += alpha/dsputil_alpha_asm.lo alpha/motion_est_mvi_asm.lo
 CFLAGS += -fforce-addr -freduce-all-givs
 endif
 
 ifeq ($(TARGET_ARCH_POWERPC),yes)
-OBJS += ppc/dsputil_ppc.o ppc/mpegvideo_ppc.o
+OBJS += ppc/dsputil_ppc.lo ppc/mpegvideo_ppc.lo
 endif
 
 ifeq ($(TARGET_MMI),yes)
-OBJS += ps2/dsputil_mmi.o ps2/idct_mmi.o ps2/mpegvideo_mmi.o
+OBJS += ps2/dsputil_mmi.lo ps2/idct_mmi.lo ps2/mpegvideo_mmi.lo
 endif
 
 ifeq ($(TARGET_ALTIVEC),yes)
-OBJS += ppc/dsputil_altivec.o ppc/mpegvideo_altivec.o ppc/idct_altivec.o \
-        ppc/fft_altivec.o ppc/gmc_altivec.o
+OBJS += ppc/dsputil_altivec.lo ppc/mpegvideo_altivec.lo ppc/idct_altivec.lo \
+        ppc/fft_altivec.lo ppc/gmc_altivec.lo
 endif
 
 ifeq ($(TARGET_ARCH_SH4),yes)
-OBJS+= sh4/idct_sh4.o sh4/dsputil_sh4.o sh4/dsputil_align.o
+OBJS+= sh4/idct_sh4.lo sh4/dsputil_sh4.lo sh4/dsputil_align.lo
 endif
 
 
-SRCS := $(OBJS:.o=.c) $(ASM_OBJS:.o=.S)
+SRCS := $(OBJS:.lo=.c) $(ASM_OBJS:.lo=.S)
 OBJS := $(OBJS) $(ASM_OBJS)
 
-LIB= $(LIBPREF)avcodec$(LIBSUF)
+LIB= libavcodec.la
 ifeq ($(BUILD_SHARED),yes)
-SLIB= $(SLIBPREF)avcodec$(SLIBSUF)
+RPATH= -rpath $(prefix)/lib
+else
+RPATH=
 endif
 TESTS= imgresample-test dct-test motion-test fft-test
 
@@ -153,28 +155,18 @@
 tests: apiexample cpuid_test $(TESTS)
 
 $(LIB): $(OBJS) $(AMRLIBS)
-	rm -f $@
-	$(AR) rc $@ $(OBJS) $(AMREXTRALIBS)
-	$(RANLIB) $@
-
-$(SLIB): $(OBJS)
-ifeq ($(CONFIG_WIN32),yes)
-	$(CC) $(SHFLAGS) -Wl,--output-def,$(@:.dll=.def) -o $@ $(OBJS) $(EXTRALIBS) $(AMREXTRALIBS)
-	-lib /machine:i386 /def:$(@:.dll=.def)
-else
-	$(CC) $(SHFLAGS) -o $@ $(OBJS) $(EXTRALIBS) $(AMREXTRALIBS)
-endif
+	libtool --mode=link $(CC) $(LDFLAGS) -o $@ $(OBJS) $(AMREXTRALIBS) $(RPATH) $(EXTRALIBS) -release $(VERSION)
 
-dsputil.o: dsputil.c dsputil.h
+dsputil.lo: dsputil.c dsputil.h
 
-libpostproc/libpostproc.a:
+libpostproc/libpostproc.la:
 	$(MAKE) -C libpostproc
 
-%.o: %.c
-	$(CC) $(CFLAGS) -c -o $@ $< 
+%.lo: %.c
+	libtool --mode=compile $(CC) $(CFLAGS) -c -o $@ $< 
 
-%.o: %.S
-	$(CC) $(CFLAGS) -c -o $@ $<
+%.o: %.S
+	libtool --mode=compile --tag CC $(CC) $(CFLAGS) -c -o $@ $<
 
 depend: $(SRCS)
 	$(CC) -MM $(CFLAGS) $^ 1>.depend
@@ -182,14 +174,14 @@
 dep:	depend
 
 clean: $(CLEANAMR)
-	rm -f *.o *.d *~ .depend $(LIB) $(SLIB) *.so i386/*.o i386/*~ \
-	   armv4l/*.o armv4l/*~ \
-	   mlib/*.o mlib/*~ \
-	   alpha/*.o alpha/*~ \
-	   ppc/*.o ppc/*~ \
-	   ps2/*.o ps2/*~ \
-	   sh4/*.o sh4/*~ \
-	   liba52/*.o liba52/*~ \
+	rm -f *.*o *.d *~ .depend $(LIB) $(SLIB) *.so* *.la i386/*.*o i386/*~ \
+	   armv4l/*.*o armv4l/*~ \
+	   mlib/*.*o mlib/*~ \
+	   alpha/*.*o alpha/*~ \
+	   ppc/*.*o ppc/*~ \
+	   ps2/*.*o ps2/*~ \
+	   sh4/*.*o sh4/*~ \
+	   liba52/*.*o liba52/*~ \
 	   apiexample $(TESTS)
 	$(MAKE) -C libpostproc clean
 
@@ -200,51 +192,49 @@
 	$(MAKE) -C amr clean
 
 cleanamrfloat:
-	rm -f amr_float/*.o
+	rm -f amr_float/*.*o
 
 # api example program
 apiexample: apiexample.c $(LIB)
-	$(CC) $(CFLAGS) -o $@ $< $(LIB) $(EXTRALIBS) -lm
+	libtool --mode=link $(CC) $(CFLAGS) -o $@ $< $(LIB) $(EXTRALIBS) -lm
 
 # cpuid test
 cpuid_test: i386/cputest.c
-	$(CC) $(CFLAGS) -D__TEST__ -o $@ $<
+	libtool --mode=link $(CC) $(CFLAGS) -D__TEST__ -o $@ $<
 
 # testing progs
 
 imgresample-test: imgresample.c
-	$(CC) $(CFLAGS) -DTEST -o $@ $^ -lm
+	libtool --mode=link $(CC) $(CFLAGS) -DTEST -o $@ $^ -lm
 
 dct-test: dct-test.o fdctref.o $(LIB)
-	$(CC) -o $@ $^ -lm
+	libtool --mode=link $(CC) -o $@ $^ -lm
 
 motion-test: motion_test.o $(LIB)
-	$(CC) -o $@ $^ -lm
+	libtool --mode=link $(CC) -o $@ $^ -lm
 
 fft-test: fft-test.o $(LIB)
-	$(CC) -o $@ $^ -lm
+	libtool --mode=link $(CC) -o $@ $^ -lm
 
 ifeq ($(BUILD_SHARED),yes)
 install: all install-headers
 ifeq ($(CONFIG_WIN32),yes)
 	install -s -m 755 $(SLIB) "$(prefix)"
 else
-	install -d $(prefix)/lib
-	install -s -m 755 $(SLIB) $(prefix)/lib/libavcodec-$(VERSION).so
-	ln -sf libavcodec-$(VERSION).so $(prefix)/lib/libavcodec.so
-	ldconfig || true
+	install -d $(DESTDIR)$(prefix)/lib
+	libtool --mode=install install $(LIB) $(DESTDIR)$(prefix)/lib
 endif
 else
 install:
 endif
 
 installlib: all install-headers
-	install -m 644 $(LIB) $(prefix)/lib
+	libtool --mode=install install $(LIB) $(DESTDIR)$(prefix)/lib
 
 install-headers:
-	mkdir -p "$(prefix)/include/ffmpeg"
+	install -d "$(DESTDIR)$(prefix)/include/ffmpeg"
 	install -m 644 $(SRC_PATH)/libavcodec/avcodec.h $(SRC_PATH)/libavcodec/common.h \
-                "$(prefix)/include/ffmpeg"
+                "$(DESTDIR)$(prefix)/include/ffmpeg"
 
 #
 # include dependency files if they exist
diff -Nur ffmpeg-0.4.8.orig/libavformat/Makefile ffmpeg-0.4.8/libavformat/Makefile
--- ffmpeg-0.4.8.orig/libavformat/Makefile	2003-09-28 17:26:40.000000000 +0200
+++ ffmpeg-0.4.8/libavformat/Makefile	2003-10-13 22:32:23.271927160 +0200
@@ -8,81 +8,71 @@
 
 CFLAGS= $(OPTFLAGS) -Wall -I.. -I$(SRC_PATH) -I$(SRC_PATH)/libavcodec -DHAVE_AV_CONFIG_H -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_GNU_SOURCE
 
-OBJS= utils.o cutils.o os_support.o allformats.o
+OBJS= utils.lo cutils.lo os_support.lo allformats.lo
 PPOBJS=
 
 # mux and demuxes
-OBJS+=mpeg.o mpegts.o mpegtsenc.o ffm.o crc.o img.o raw.o rm.o \
-      avienc.o avidec.o wav.o swf.o au.o gif.o mov.o mpjpeg.o dvcore.o dv.o \
-      yuv4mpeg.o 4xm.o flvenc.o flvdec.o movenc.o psxstr.o idroq.o ipmovie.o \
-      nut.o wc3movie.o mp3.o
+OBJS+=mpeg.lo mpegts.lo mpegtsenc.lo ffm.lo crc.lo img.lo raw.lo rm.lo \
+      avienc.lo avidec.lo wav.lo swf.lo au.lo gif.lo mov.lo mpjpeg.lo dvcore.lo dv.lo \
+      yuv4mpeg.lo 4xm.lo flvenc.lo flvdec.lo movenc.lo psxstr.lo idroq.lo ipmovie.lo \
+      nut.lo wc3movie.lo mp3.lo
 
 ifeq ($(CONFIG_RISKY),yes)
-OBJS+= asf.o
+OBJS+= asf.lo
 endif
 
 ifeq ($(AMR_NB),yes)
-OBJS+= amr.o
+OBJS+= amr.lo
 endif
 
 # image formats
-OBJS+= pnm.o yuv.o png.o jpeg.o gifdec.o
+OBJS+= pnm.lo yuv.lo png.lo jpeg.lo gifdec.lo
 # file I/O
-OBJS+= avio.o aviobuf.o file.o 
-OBJS+= framehook.o 
+OBJS+= avio.lo aviobuf.lo file.lo
+OBJS+= framehook.lo
 
 ifeq ($(CONFIG_VIDEO4LINUX),yes)
-OBJS+= grab.o
+OBJS+= grab.lo
 endif
 
 ifeq ($(CONFIG_DV1394),yes)
-OBJS+= dv1394.o
+OBJS+= dv1394.lo
 endif
 
 ifeq ($(CONFIG_AUDIO_OSS),yes)
-OBJS+= audio.o 
+OBJS+= audio.lo
 endif
 
 ifeq ($(CONFIG_AUDIO_BEOS),yes)
-PPOBJS+= beosaudio.o
+PPOBJS+= beosaudio.lo
 EXTRALIBS+=-lbe -lmedia
-# this should be the default !
-EXTRALIBS+=-lavcodec -L../libavcodec
 endif
 
 ifeq ($(CONFIG_NETWORK),yes)
-OBJS+= udp.o tcp.o http.o rtsp.o rtp.o rtpproto.o
+OBJS+= udp.lo tcp.lo http.lo rtsp.lo rtp.lo rtpproto.lo
 # BeOS and Darwin network stuff
 ifeq ($(NEED_INET_ATON),yes)
-OBJS+= barpainet.o
+OBJS+= barpainet.lo
 endif
 endif
 
 ifeq ($(CONFIG_VORBIS),yes)
-OBJS+= ogg.o
+OBJS+= ogg.lo
 endif
 
-LIB= $(LIBPREF)avformat$(LIBSUF)
+LIB= libavformat.la
 ifeq ($(BUILD_SHARED),yes)
-SLIB= $(SLIBPREF)avformat$(SLIBSUF)
+RPATH= -rpath $(prefix)/lib
+else
+RPATH=
 endif
 
-SRCS := $(OBJS:.o=.c) $(PPOBJS:.o=.cpp)
+SRCS := $(OBJS:.lo=.c) $(PPOBJS:.lo=.cpp)
 
 all: $(LIB) $(SLIB)
 
 $(LIB): $(OBJS) $(PPOBJS)
-	rm -f $@
-	$(AR) rc $@ $(OBJS) $(PPOBJS)
-	$(RANLIB) $@
-
-$(SLIB): $(OBJS)
-ifeq ($(CONFIG_WIN32),yes)
-	$(CC) $(SHFLAGS) -Wl,--output-def,$(@:.dll=.def) -o $@ $(OBJS) $(PPOBJS) $(EXTRALIBS) $(VPATH)/../libavcodec/avcodec.dll
-	-lib /machine:i386 /def:$(@:.dll=.def)
-else
-	$(CC) $(SHFLAGS) -o $@ $(OBJS) $(PPOBJS) $(EXTRALIBS)
-endif
+	libtool --mode=link $(CC) $(LDOPT) -o $@ $(OBJS) $(PPOBJS) ../libavcodec/libavcodec.la $(RPATH) -release $(VERSION)
 
 depend: $(SRCS)
 	$(CC) -MM $(CFLAGS) $^ 1>.depend
@@ -92,34 +82,32 @@
 ifeq ($(CONFIG_WIN32),yes)
 	install -s -m 755 $(SLIB) "$(prefix)"
 else
-	install -d $(prefix)/lib
-	install -s -m 755 $(SLIB) $(prefix)/lib/libavformat-$(VERSION).so
-	ln -sf libavformat-$(VERSION).so $(prefix)/lib/libavformat.so
-	ldconfig || true
+	install -d $(DESTDIR)$(prefix)/lib
+	libtool --mode=install install $(LIB) $(DESTDIR)$(prefix)/lib
 endif
 else
 install:
 endif
 
 installlib: all install-headers
-	install -m 644 $(LIB) $(prefix)/lib
+	libtool --mode=install install $(LIB) $(DESTDIR)$(prefix)/lib
 
 install-headers:
-	mkdir -p "$(prefix)/include/ffmpeg"
+	install -d "$(DESTDIR)$(prefix)/include/ffmpeg"
 	install -m 644 $(SRC_PATH)/libavformat/avformat.h $(SRC_PATH)/libavformat/avio.h \
                 $(SRC_PATH)/libavformat/rtp.h $(SRC_PATH)/libavformat/rtsp.h \
                 $(SRC_PATH)/libavformat/rtspcodes.h \
-                "$(prefix)/include/ffmpeg"
+                "$(DESTDIR)$(prefix)/include/ffmpeg"
 
-%.o: %.c
-	$(CC) $(CFLAGS) -c -o $@ $< 
+%.lo: %.c
+	libtool --mode=compile $(CC) $(CFLAGS) -c -o $@ $< 
 
 # BeOS: remove -Wall to get rid of all the "multibyte constant" warnings
 %.o: %.cpp
-	g++ $(subst -Wall,,$(CFLAGS)) -c -o $@ $< 
+	libtool --mode=compile --tag CXX g++ $(subst -Wall,,$(CFLAGS)) -c -o $@ $< 
 
 clean: 
-	rm -f *.o *.d .depend *~ *.a $(LIB)
+	rm -f *.*o *.d .depend *~ *.*a $(LIB)
 
 #
 # include dependency files if they exist
diff -Nur ffmpeg-0.4.8.orig/Makefile ffmpeg-0.4.8/Makefile
--- ffmpeg-0.4.8.orig/Makefile	2003-09-28 17:26:39.000000000 +0200
+++ ffmpeg-0.4.8/Makefile	2003-10-13 22:37:04.668148432 +0200
@@ -29,14 +29,10 @@
 EXTRALIBS+=-lmedia -lbe
 endif
 
-ifeq ($(BUILD_SHARED),yes)
-DEP_LIBS=libavcodec/$(SLIBPREF)avcodec$(SLIBSUF) libavformat/$(SLIBPREF)avformat$(SLIBSUF)
-else
-DEP_LIBS=libavcodec/$(LIBPREF)avcodec$(LIBSUF) libavformat/$(LIBPREF)avformat$(LIBSUF)
+DEP_LIBS=libavcodec/libavcodec.la libavformat/libavformat.la
 ifeq ($(CONFIG_MP3LAME),yes)
 EXTRALIBS+=-lmp3lame
 endif
-endif
 
 ifeq ($(CONFIG_VORBIS),yes)
 EXTRALIBS+=-logg -lvorbis -lvorbisenc
@@ -64,7 +60,7 @@
 
 OBJS = ffmpeg.o ffserver.o cmdutils.o ffplay.o
 SRCS = $(OBJS:.o=.c) $(ASM_OBJS:.o=.s)
-FFLIBS = -L./libavformat -lavformat -L./libavcodec -lavcodec
+FFLIBS = libavformat/libavformat.la libavcodec/libavcodec.la
 
 all: lib $(PROG) $(PROGTEST) $(VHOOK)
 
@@ -72,25 +68,17 @@
 	$(MAKE) -C libavcodec all
 	$(MAKE) -C libavformat all
 
-ffmpeg_g$(EXESUF): ffmpeg.o cmdutils.o .libs
-	$(CC) $(LDFLAGS) -o $@ ffmpeg.o cmdutils.o $(FFLIBS) $(EXTRALIBS)
+ffmpeg$(EXESUF): ffmpeg.o cmdutils.o
+	libtool --mode=link $(CC) $(LDFLAGS) -o $@ ffmpeg.o cmdutils.o $(FFLIBS) $(EXTRALIBS)
 
-ffmpeg$(EXESUF): ffmpeg_g$(EXESUF)
-	cp -p $< $@
-	$(STRIP) $@
-
-ffserver$(EXESUF): ffserver.o .libs
-	$(CC) $(LDFLAGS) $(FFSLDFLAGS) -o $@ ffserver.o $(FFLIBS) $(EXTRALIBS) 
-
-ffplay_g$(EXESUF): ffplay.o cmdutils.o .libs
-	$(CC) $(LDFLAGS) -o $@ ffplay.o cmdutils.o $(FFLIBS) $(EXTRALIBS) $(SDL_LIBS)
-
-ffplay$(EXESUF): ffplay_g$(EXESUF)
-	cp -p $< $@
-	$(STRIP) $@
+ffserver$(EXESUF): ffserver.o
+	libtool --mode=link $(CC) $(LDFLAGS) $(FFSLDFLAGS) -o $@ ffserver.o $(FFLIBS) $(EXTRALIBS)
+
+ffplay$(EXESUF): ffplay.o cmdutils.o
+	libtool --mode=link $(CC) $(LDFLAGS) -o $@ ffplay.o cmdutils.o $(FFLIBS) $(EXTRALIBS) $(SDL_LIBS)
 
 output_example$(EXESUF): output_example.o .libs
-	$(CC) $(LDFLAGS) -o $@ output_example.o $(FFLIBS) $(EXTRALIBS)
+	libtool --mode=link $(CC) $(LDFLAGS) -o $@ output_example.o $(FFLIBS) $(EXTRALIBS)
 
 ffplay.o: ffplay.c
 	$(CC) $(CFLAGS) $(SDL_CFLAGS) -c -o $@ $< 
@@ -98,16 +86,16 @@
 %.o: %.c
 	$(CC) $(CFLAGS) -c -o $@ $< 
 
-videohook: .libs
+videohook:
 	$(MAKE) -C vhook all
 
 .PHONY: install
 
-install: all install-man $(INSTALLVHOOK)
+install: install-man $(INSTALLVHOOK)
 	$(MAKE) -C libavcodec install
 	$(MAKE) -C libavformat install
-	install -d "$(bindir)"
-	install -c -s -m 755 $(PROG) "$(bindir)"
+	install -d "$(DESTDIR)$(bindir)"
+	libtool --mode=install install -c -m 755 $(PROG) "$(DESTDIR)$(bindir)"
 
 # create the window installer
 wininstaller: all install
@@ -117,16 +105,16 @@
 install-man:
 ifneq ($(CONFIG_WIN32),yes)
 	if [ -f $(SRC_PATH)/doc/ffmpeg.1 ] ; then \
-	    install -d $(mandir)/man1 ; \
+	    install -d $(DESTDIR)$(mandir)/man1 ; \
 	    install -m 644 $(SRC_PATH)/doc/ffmpeg.1 $(SRC_PATH)/doc/ffplay.1 \
-                           $(SRC_PATH)/doc/ffserver.1 $(mandir)/man1 ; \
+                           $(SRC_PATH)/doc/ffserver.1 $(DESTDIR)$(mandir)/man1 ; \
 	fi
 endif
 
-install-vhook: $(prefix)/lib/vhook
-	$(MAKE) -C vhook install INSTDIR=$(prefix)/lib/vhook
+install-vhook: $(DESTDIR)$(prefix)/lib/vhook
+	$(MAKE) -C vhook install INSTDIR=$(DESTDIR)$(prefix)/lib/vhook
 
-$(prefix)/lib/vhook:
+$(DESTDIR)$(prefix)/lib/vhook:
 	install -d $@
 
 installlib:
@@ -145,15 +133,11 @@
 .depend: $(SRCS)
 	$(CC) -MM $(CFLAGS) $^ 1>.depend
 
-.libs: lib
-	@test -f .libs || touch .libs
-	@for i in $(DEP_LIBS) ; do if $(TEST) $$i -nt .libs ; then touch .libs; fi ; done
-
 clean: $(CLEANVHOOK)
 	$(MAKE) -C libavcodec clean
 	$(MAKE) -C libavformat clean
 	$(MAKE) -C tests clean
-	rm -f *.o *.d *~ .libs .depend gmon.out TAGS ffmpeg_g$(EXESUF) ffplay_g$(EXESUF) $(PROG) $(PROGTEST)
+	rm -f *.o *.d *~ .depend gmon.out TAGS $(PROG) $(PROGTEST)
 
 clean-vhook:
 	$(MAKE) -C vhook clean
