--- ffmpeg-export-2006-12-04/configure.orig	2006-12-09 16:02:48.392192000 +0100
+++ ffmpeg-export-2006-12-04/configure	2006-12-09 17:22:37.765123062 +0100
@@ -1891,6 +1891,25 @@
 lavu_version=`grep '#define LIBAVUTIL_VERSION ' "$source_path/libavutil/avutil.h" | sed 's/[^0-9\.]//g'`
 
 
+if test "$amr_wb" = "yes" -o "$amr_nb" = "yes" ; then
+  echo "CONFIG_AMR=yes" >> config.mak
+fi
+
+if test "$amr_wb" = "yes" ; then
+  echo "#define AMR_WB 1" >> $TMPH
+  echo "CONFIG_AMR_WB=yes" >> config.mak
+  extralibs="$extralibs -lamrwb"
+fi
+
+if test "$amr_nb" = "yes" ; then
+  echo "#define AMR_NB 1" >> $TMPH
+  echo "CONFIG_AMR_NB=yes" >> config.mak
+if test "$amr_nb_fixed" = "yes" ; then
+  extralibs="$extralibs -lamrnbfixed"
+else
+  extralibs="$extralibs -lamrnb"
+fi
+fi
 
 if test "$lshared" = "yes" ; then
   echo "#define BUILD_SHARED_AV 1" >> $TMPH
--- ffmpeg-export-2006-12-04/libavcodec/Makefile.orig	2006-11-22 10:20:16.000000000 +0100
+++ ffmpeg-export-2006-12-04/libavcodec/Makefile	2006-12-09 17:23:09.810949249 +0100
@@ -264,33 +264,6 @@
 OBJS-$(CONFIG_DTS)                     += dtsdec.o
 
 OBJS-$(CONFIG_AMR)                     += amr.o
-OBJS-$(CONFIG_AMR_NB)                  += amr_float/sp_dec.o     \
-                                          amr_float/sp_enc.o     \
-                                          amr_float/interf_dec.o \
-                                          amr_float/interf_enc.o
-
-ifeq ($(CONFIG_AMR_NB_FIXED),yes)
-EXTRAOBJS += amr/*.o
-EXTRADEPS=amrlibs
-endif
-
-OBJS-$(CONFIG_AMR_WB)                  += amrwb_float/dec_acelp.o \
-                                          amrwb_float/dec_dtx.o   \
-                                          amrwb_float/dec_gain.o  \
-                                          amrwb_float/dec_if.o    \
-                                          amrwb_float/dec_lpc.o   \
-                                          amrwb_float/dec_main.o  \
-                                          amrwb_float/dec_rom.o   \
-                                          amrwb_float/dec_util.o  \
-                                          amrwb_float/enc_acelp.o \
-                                          amrwb_float/enc_dtx.o   \
-                                          amrwb_float/enc_gain.o  \
-                                          amrwb_float/enc_if.o    \
-                                          amrwb_float/enc_lpc.o   \
-                                          amrwb_float/enc_main.o  \
-                                          amrwb_float/enc_rom.o   \
-                                          amrwb_float/enc_util.o  \
-                                          amrwb_float/if_rom.o
 
 OBJS-$(CONFIG_AAC_PARSER)              += parser.o
 OBJS-$(CONFIG_AC3_PARSER)              += parser.o
--- ffmpeg-export-2006-12-04/libavcodec/amr.c.orig	2006-11-20 21:41:31.000000000 +0100
+++ ffmpeg-export-2006-12-04/libavcodec/amr.c	2006-12-09 17:24:13.690589543 +0100
@@ -61,16 +61,16 @@
 
 #define MMS_IO
 
-#include "amr/sp_dec.h"
-#include "amr/d_homing.h"
-#include "amr/typedef.h"
-#include "amr/sp_enc.h"
-#include "amr/sid_sync.h"
-#include "amr/e_homing.h"
+#include <amr/sp_dec.h>
+#include <amr/d_homing.h>
+#include <amr/typedef.h>
+#include <amr/sp_enc.h>
+#include <amr/sid_sync.h>
+#include <amr/e_homing.h>
 
 #else
-#include "amr_float/interf_dec.h"
-#include "amr_float/interf_enc.h"
+#include <amrnb/interf_dec.h>
+#include <amrnb/interf_enc.h>
 #endif
 
 /* Common code for fixed and float version*/
@@ -532,8 +532,8 @@
 #define typedef_h
 #endif
 
-#include "amrwb_float/enc_if.h"
-#include "amrwb_float/dec_if.h"
+#include <amrwb/enc_if.h>
+#include <amrwb/dec_if.h>
 
 /* Common code for fixed and float version*/
 typedef struct AMRWB_bitrates
