--- ffmpeg-0.8/libavutil/avutil.h~	2011-06-21 22:08:33.000000000 +0200
+++ ffmpeg-0.8/libavutil/avutil.h	2011-07-26 22:35:36.646420640 +0200
@@ -104,7 +104,8 @@
 #define AV_TIME_BASE_Q          (AVRational){1, AV_TIME_BASE}
 
 enum AVPictureType {
-    AV_PICTURE_TYPE_I = 1, ///< Intra
+    AV_PICTURE_TYPE_NONE = 0, ///< Undefined
+    AV_PICTURE_TYPE_I,     ///< Intra
     AV_PICTURE_TYPE_P,     ///< Predicted
     AV_PICTURE_TYPE_B,     ///< Bi-dir predicted
     AV_PICTURE_TYPE_S,     ///< S(GMC)-VOP MPEG4
