--- ffmpeg-0.4.7/vhook/Makefile.orig	2003-08-25 00:29:51.000000000 +0200
+++ ffmpeg-0.4.7/vhook/Makefile	2003-09-04 19:47:04.081085776 +0200
@@ -26,16 +26,16 @@
 	$(CC) -MM $(CFLAGS) $^ 1>.depend
 
 install:
-	install -s -m 755 $(HOOKS) $(INSTDIR)
+	install -m 755 $(HOOKS) $(INSTDIR)
 
 imlib2.so: imlib2.o
-	$(CC) -g -o $@ $(SHFLAGS) $< -lImlib2
+	$(CC) $(LDOPT) -o $@ $(SHFLAGS) $< -lImlib2
 
 drawtext.so: drawtext.o
-	$(CC) -g -o $@ $(SHFLAGS) $< `freetype-config --libs`
+	$(CC) $(LDOPT) -o $@ $(SHFLAGS) $< `freetype-config --libs`
 
 %.so: %.o
-	$(CC) -g -o $@ $(SHFLAGS) $<
+	$(CC) $(LDOPT) -o $@ $(SHFLAGS) $<
 
 clean:
 	rm -f *.o *.d .depend *.so *~
