--- ffmpeg/configure.orig	2006-09-18 13:10:15.000000000 +0000
+++ ffmpeg/configure	2006-09-18 13:39:55.760264500 +0000
@@ -1026,7 +1026,7 @@
 EOF
 
 have_lrintf="no"
-if $cc $CFLAGS $LDFLAGS $extralibs -o $TMPE $TMPC 2> /dev/null ; then
+if $cc $CFLAGS $LDFLAGS -o $TMPE $TMPC $extralibs 2> /dev/null ; then
   have_lrintf="yes"
   # allanc@chickenandporn.com: cannot execute cross-compiled
   # code on the host.  Only execute if not cross-compiling.
