--- ffmpeg-export-2006-12-04.orig/libavcodec/ppc/h264_altivec.c	2006-12-02 14:50:42.000000000 +0000
+++ ffmpeg-export-2006-12-04.altivec_fix/libavcodec/ppc/h264_altivec.c	2006-12-06 04:58:41.000000000 +0000
@@ -464,7 +464,7 @@
     vec_u8_t lv = vec_ld( 7, dest );                           \
     vec_u8_t dstv   = vec_perm( hv, lv, (vec_u8_t)perm_ldv );  \
     vec_s16_t idct_sh6 = vec_sra(idctv, sixv);                 \
-    vec_u16_t dst16 = vec_mergeh(zero_u8v, dstv);              \
+    vec_u16_t dst16 = (vec_u16_t)vec_mergeh(zero_u8v, dstv);   \
     vec_s16_t idstsum = vec_adds(idct_sh6, (vec_s16_t)dst16);  \
     vec_u8_t idstsum8 = vec_packsu(zero_s16v, idstsum);        \
     vec_u8_t edgehv;                                           \
