--- ffmpeg-0.11.2/configure.orig	2012-10-05 21:04:40.855184264 +0200
+++ ffmpeg-0.11.2/configure	2012-10-05 21:23:33.891827315 +0200
@@ -3225,7 +3225,7 @@
 enabled libopencore_amrnb  && require libopencore_amrnb opencore-amrnb/interf_dec.h Decoder_Interface_init -lopencore-amrnb
 enabled libopencore_amrwb  && require libopencore_amrwb opencore-amrwb/dec_if.h D_IF_init -lopencore-amrwb
 enabled libopencv  && add_extralibs -lopencv_core -lopencv_features2d -lopencv_imgproc -lopencv_objdetect -lgcrypt
-enabled libopenjpeg && require libopenjpeg openjpeg.h opj_version -lopenjpeg
+enabled libopenjpeg && require_pkg_config libopenjpeg openjpeg.h opj_version
 enabled libpulse && require_pkg_config libpulse-simple pulse/simple.h pa_simple_new
 enabled librtmp    && require_pkg_config librtmp librtmp/rtmp.h RTMP_Socket
 enabled libschroedinger && require_pkg_config schroedinger-1.0 schroedinger/schro.h schro_init
