--- ffmpeg-export-2007-06-26/vhook/Makefile~	2007-06-14 19:43:56.000000000 +0300
+++ ffmpeg-export-2007-06-26/vhook/Makefile	2007-06-27 00:08:42.436953372 +0300
@@ -9,9 +9,9 @@
 HOOKS=null$(SLIBSUF) fish$(SLIBSUF) watermark$(SLIBSUF)
 ALLHOOKS=$(HOOKS) ppm$(SLIBSUF) imlib2$(SLIBSUF) drawtext$(SLIBSUF)
 
-ifeq ($(HAVE_FORK),yes)
+#ifeq ($(HAVE_FORK),yes)
     HOOKS += ppm$(SLIBSUF)
-endif
+#endif
 
 ifeq ($(HAVE_IMLIB2),yes)
     HOOKS += imlib2$(SLIBSUF)
