--- ffmpeg-1.0.1/libavdevice/libcdio.c.orig	2012-12-03 22:17:35.000000000 +0100
+++ ffmpeg-1.0.1/libavdevice/libcdio.c	2012-12-06 19:03:32.695282861 +0100
@@ -23,8 +23,8 @@
  * libcdio CD grabbing
  */
 
-#include <cdio/cdda.h>
-#include <cdio/paranoia.h>
+#include <cdio/paranoia/cdda.h>
+#include <cdio/paranoia/paranoia.h>
 
 #include "libavutil/log.h"
 #include "libavutil/mem.h"
--- ffmpeg-1.1/configure.orig	2013-01-12 20:17:28.423418802 +0100
+++ ffmpeg-1.1/configure	2013-01-12 20:41:48.743388367 +0100
@@ -3817,7 +3817,7 @@
 enabled libaacplus && require  "libaacplus >= 2.0.0" aacplus.h aacplusEncOpen -laacplus
 enabled libass     && require_pkg_config libass ass/ass.h ass_library_init
 enabled libbluray  && require libbluray libbluray/bluray.h bd_open -lbluray
-enabled libcdio    && require2 libcdio "cdio/cdda.h cdio/paranoia.h" cdio_cddap_open -lcdio_paranoia -lcdio_cdda -lcdio
+enabled libcdio    && require2 libcdio "cdio/paranoia/cdda.h cdio/paranoia/paranoia.h" cdio_cddap_open -lcdio_paranoia -lcdio_cdda -lcdio
 enabled libcelt    && require libcelt celt/celt.h celt_decode -lcelt0 &&
                       { check_lib celt/celt.h celt_decoder_create_custom -lcelt0 ||
                         die "ERROR: libcelt must be installed and version must be >= 0.11.0."; }
