--- ffmpeg-export-2007-06-26/configure.orig	2007-06-25 18:18:28.000000000 +0200
+++ ffmpeg-export-2007-06-26/configure	2007-08-27 20:49:39.953203827 +0200
@@ -1574,7 +1574,9 @@
 enabled_any libamr_nb libamr_wb && enable libamr
 
 # these are off by default, so fail if requested and not available
+save_extralibs="$extralibs"
 enabled liba52     && require liba52 a52dec/a52.h a52_init -la52
+enabled liba52bin  && extralibs="$save_extralibs"
 enabled libamr_nb  && require libamrnb amrnb/interf_dec.h Speech_Decode_Frame_init -lamrnb -lm
 enabled libamr_wb  && require libamrwb amrwb/dec_if.h D_IF_init -lamrwb -lm
 enabled libgsm     && require libgsm gsm.h gsm_create -lgsm
