--- ffmpeg-0.10/libavcodec/libmp3lame.c.orig	2012-01-26 23:15:59.000000000 +0100
+++ ffmpeg-0.10/libavcodec/libmp3lame.c	2012-01-27 17:53:34.901932031 +0100
@@ -38,6 +38,7 @@
     int stereo;
     uint8_t buffer[BUFFER_SIZE];
     int buffer_index;
+    int flushed;
     struct {
         int *left;
         int *right;
@@ -181,11 +182,16 @@
     /* lame 3.91 dies on '1-channel interleaved' data */
 
     if (!data){
+	if(s->flushed)
+	    lame_result = 0;
+	else {
+	    s->flushed = 1;
         lame_result= lame_encode_flush(
                 s->gfp,
                 s->buffer + s->buffer_index,
                 BUFFER_SIZE - s->buffer_index
                 );
+	}
 #if 2147483647 == INT_MAX
     }else if(AV_SAMPLE_FMT_S32 == avctx->sample_fmt){
         if (s->stereo) {
@@ -217,6 +223,7 @@
                 BUFFER_SIZE - s->buffer_index
                 );
         }
+	s->flushed = 0;
 #endif
     }else{
         if (s->stereo) {
@@ -237,6 +244,7 @@
                 BUFFER_SIZE - s->buffer_index
                 );
         }
+	s->flushed = 0;
     }
 
     if (lame_result < 0) {
