--- ffmpeg-0.4.6/Makefile.orig	Fri Dec 27 15:35:45 2002
+++ ffmpeg-0.4.6/Makefile	Sun Jan  5 01:23:59 2003
@@ -6,8 +6,8 @@
 
 VPATH=$(SRC_PATH)
 
-CFLAGS= $(OPTFLAGS) -Wall -g -I. -I$(SRC_PATH) -I$(SRC_PATH)/libavcodec -I$(SRC_PATH)/libavformat -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_GNU_SOURCE
-LDFLAGS+= -g 
+CFLAGS= $(OPT) -Wall -I. -I$(SRC_PATH) -I$(SRC_PATH)/libavcodec -I$(SRC_PATH)/libavformat -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_GNU_SOURCE
+LDFLAGS+= $(LDOPT)
 
 ifeq ($(TARGET_GPROF),yes)
 CFLAGS+=-p
--- ffmpeg-0.4.6/libavcodec/Makefile.orig	Fri Dec 27 15:35:45 2002
+++ ffmpeg-0.4.6/libavcodec/Makefile	Sun Jan  5 01:24:39 2003
@@ -7,8 +7,8 @@
 VPATH=$(SRC_PATH)/libavcodec
 
 # NOTE: -I.. is needed to include config.h
-CFLAGS= $(OPTFLAGS) -Wall -g -DHAVE_AV_CONFIG_H -I.. -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_GNU_SOURCE
-LDFLAGS= -g
+CFLAGS= $(OPT) -Wall -DHAVE_AV_CONFIG_H -I.. -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_GNU_SOURCE
+LDFLAGS= $(LDOPT)
 
 OBJS= common.o utils.o mem.o allcodecs.o \
       mpegvideo.o h263.o jrevdct.o jfdctfst.o jfdctint.o\
--- ffmpeg-0.4.6/libavformat/Makefile.orig	Fri Dec 27 15:35:47 2002
+++ ffmpeg-0.4.6/libavformat/Makefile	Sun Jan  5 01:25:07 2003
@@ -6,7 +6,7 @@
 
 VPATH=$(SRC_PATH)/libavformat
 
-CFLAGS= $(OPTFLAGS) -Wall -g -I.. -I$(SRC_PATH) -I$(SRC_PATH)/libavcodec -DHAVE_AV_CONFIG_H -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_GNU_SOURCE
+CFLAGS= $(OPT) -Wall -I.. -I$(SRC_PATH) -I$(SRC_PATH)/libavcodec -DHAVE_AV_CONFIG_H -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_GNU_SOURCE
 
 OBJS= utils.o cutils.o allformats.o
 
--- ffmpeg-0.4.7/vhook/Makefile.orig	2003-08-25 00:29:51.000000000 +0200
+++ ffmpeg-0.4.7/vhook/Makefile	2003-09-04 19:47:04.081085776 +0200
@@ -2,7 +2,7 @@
 
 VPATH=$(SRC_PATH)/vhook
 
-CFLAGS=-fPIC $(OPTFLAGS) -g -Wall -I.. -I$(SRC_PATH) -I$(SRC_PATH)/libavformat -I$(SRC_PATH)/libavcodec -DHAVE_AV_CONFIG_H
+CFLAGS=-fPIC $(OPT) -Wall -I.. -I$(SRC_PATH) -I$(SRC_PATH)/libavformat -I$(SRC_PATH)/libavcodec -DHAVE_AV_CONFIG_H
 
 ifeq ($(CONFIG_DARWIN),yes)
 SHFLAGS+=-bundle -flat_namespace -undefined suppress
@@ -26,16 +26,16 @@
 	$(CC) -MM $(CFLAGS) $^ 1>.depend
 
 install:
-	install -s -m 755 $(HOOKS) $(INSTDIR)
+	install -m 755 $(HOOKS) $(INSTDIR)
 
 imlib2.so: imlib2.o
-	$(CC) -g -o $@ $(SHFLAGS) $< -lImlib2
+	$(CC) $(LDOPT) -o $@ $(SHFLAGS) $< -lImlib2
 
 drawtext.so: drawtext.o
-	$(CC) -g -o $@ $(SHFLAGS) $< `freetype-config --libs`
+	$(CC) $(LDOPT) -o $@ $(SHFLAGS) $< `freetype-config --libs`
 
 %.so: %.o
-	$(CC) -g -o $@ $(SHFLAGS) $<
+	$(CC) $(LDOPT) -o $@ $(SHFLAGS) $<
 
 clean:
 	rm -f *.o *.d .depend *.so *~
